### PR TITLE
Add suggestions support (#977)

### LINKED ISSUE #895: "Did you mean" hint on wrong command or flag
We could tell the user what the closest command or argument match is by calculating the [levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) between them.

What to do you think?

### PR DESCRIPTION
**Pull request by meatballhat**
## What type of PR is this?

- feature

## What this PR does / why we need it:

See #977 

## Which issue(s) this PR fixes:

Closes #895 

## Special notes for your reviewer:

This is intended to carry forward the work in #977 with the addition of build tags to allow opting out of the suggestions dependency at build time as with recently-added work for markdown/man doc dependencies (cc @kolyshkin).

### CODE REVIEW

**Comment by kolyshkin** [comment]
@meatballhat what is the size difference with and without suggestions?

File: suggestions.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import (
+	"fmt"
+
+	"github.com/antzucaro/matchr"
+)
+
+const didYouMeanTemplate = "Did you mean '%s'?"
+
+func (a *App) suggestFlagFromError(err error, command string) (string, error) {
```
[the remark this answers is not in the document]
Comment (meatballhat) [reply suggestions.go:14]: Oooooh I really like that idea

**Comment by meatballhat** [comment]
@kolyshkin regarding size difference

```sh-session
$ go run ./internal/build/build.go check-binary-size
# ---> /usr/sbin/git rev-parse --show-toplevel
# ---> /home/me/.gimme/versions/go1.18.1.linux.amd64/bin/go build -tags  -o ./internal/example-cli/built-example -ldflags -s -w ./internal/example-cli/example-cli.go
# ---> /home/me/.gimme/versions/go1.18.1.linux.amd64/bin/go build -tags  -o ./internal/example-hello-world/built-example -ldflags -s -w ./internal/example-hello-world/example-hello-world.go

1.7MB is the current binary size
  ✅ 1.7MB is the target min size
  ✅ 2.2MB is the target max size
```

```sh-session
$ go run ./internal/build/build.go --tags urfave_cli_no_suggest check-binary-size
# ---> /usr/sbin/git rev-parse --show-toplevel
# ---> /home/me/.gimme/versions/go1.18.1.linux.amd64/bin/go build -tags urfave_cli_no_suggest -o ./internal/example-cli/built-example -ldflags -s -w ./internal/example-cli/example-cli.go
# ---> /home/me/.gimme/versions/go1.18.1.linux.amd64/bin/go build -tags urfave_cli_no_suggest -o ./internal/example-hello-world/built-example -ldflags -s -w ./internal/example-hello-world/example-hello-world.go

1.7MB is the current binary size
  ✅ 1.7MB is the target min size
  ✅ 2.2MB is the target max size
```

(small enough that it's not showing up?)

```sh-session
$ du internal/example-cli/built-example*
2852    internal/example-cli/built-example
2844    internal/example-cli/built-example-no-suggest
```

File: suggestions.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (meatballhat) [reply suggestions.go:14]: @dearchap How about override capability in a follow-up PR?

[the remark this answers is not in the document]
Comment (meatballhat) [reply suggestions.go:14]: e.g. https://github.com/urfave/cli/compare/urfave:9bd6349...urfave:34ee6c1

[the remark this answers is not in the document]
Comment (dearchap) [reply suggestions.go:14]: Nice.

**Comment by kolyshkin** [comment]
> (small enough that it's not showing up?)

The size difference seems to be about 0.1% (a few kilobytes), so it probably does not make sense to add a build tag for this.

**Pull request merged**